ui/src/kapacitor/containers/KapacitorPage.js

```diff
@@ -1,213 +1,124 @@
-                      set up alert end points like email, Slack, and others.
-                    </p>
-                    <hr/>
-                    <h4 className="text-center">Connection Details</h4>
-                    <br/>
-                    <form onSubmit={this.handleKapacitorUpdate}>
-                      <div>
-                        <div className="form-group col-xs-12 col-sm-8 col-sm-offset-2 col-md-4 col-md-offset-2">
-                          <label htmlFor="connect-string">Connection String</label>
-                          <input ref={(r) => this.kapacitorURL = r} className="form-control" id="connect-string" placeholder={defaultKapacitorUrl} value={url} onChange={this.updateURL}></input>
-                        </div>
-                        <div className="form-group col-xs-12 col-sm-8 col-sm-offset-2 col-md-4 col-md-offset-0">
-                          <label htmlFor="name">Name</label>
-                          <input ref={(r) => this.kapacitorName = r} className="form-control" id="name" placeholder={defaultKapacitorName} value={name} onChange={this.updateName}></input>
-                        </div>
-                        <div className="form-group col-xs-12 col-sm-4 col-sm-offset-2 col-md-4 col-md-offset-2">
-                          <label htmlFor="username">Username</label>
-                          <input ref={(r) => this.kapacitorUser = r} className="form-control" id="username" value={username} onChange={this.updateUsername}></input>
-                        </div>
-                        <div className="form-group col-xs-12 col-sm-4 col-md-4">
-                          <label htmlFor="password">Password</label>
-                          <input ref={(r) => this.kapacitorPassword = r} className="form-control" id="password" type="password"></input>
-                        </div>
-                      </div>
-
-                      <div className="form-group form-group-submit col-xs-12 text-center">
-                        <button className="btn btn-info" onClick={this.handleResetToDefaults}>Reset to Default</button>
-                        <button className="btn btn-success" type="submit">Connect Kapacitor</button>
-                      </div>
-                    </form>
-                  </div>
-                </div>
-              </div>
-            </div>
-            <div className="row">
-              <div className="col-md-8 col-md-offset-2">
-                {this.renderAlertOutputs()}
-              </div>
-            </div>
-          </div>
-        </div>
-      </div>
-    );
+    this.setState({kapacitor: {...defaultState}});
   },
 
-  renderAlertOutputs() {
-    const {kapacitor} = this.state;
-    if (kapacitor) {
-      return <AlertOutputs source={this.props.source} kapacitor={kapacitor} addFlashMessage={this.props.addFlashMessage} />;
-    }
+  _parseKapacitorURL() {
+    const parser = document.createElement('a');
+    parser.href = this.props.source.url;
```

Comment:
Haha, that trick. Stuff like this doesn't stand up to Node-based unit testing much, like Ava, if we were to go that route. It really does lock us into the slower Phantom tests. That said, there are no plans for an Ava port, so at this point it's no big deal.